src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -118,6 +158,20 @@ private static RepoConfiguration getMatchingRepoConfig(
         return null;
     }
 
+    /**
+     * Iterates through {@code repoConfigs} to find a {@code RepoConfiguration} with {@code RepoLocation}
+     * that matches {@code groupConfig}. Returns {@code null} if no match is found.
+     */
+    private static RepoConfiguration getMatchingRepoConfigForGroups(
+            List<RepoConfiguration> repoConfigs, GroupConfiguration groupConfig) {
+        for (RepoConfiguration repoConfig : repoConfigs) {
+            if (repoConfig.getLocation().equals(groupConfig.getLocation())) {
+                return repoConfig;
```

Comment:
This seems to find the 1st `RepoConfiguration` witht the same `RepoLocation` to match, but there may be multiple `RepoConfiguration` with the same `RepoLocation`, just with different `branch` instead right? Should those have this `GroupConfiguration` added to it too?

Reply:
`GroupConfiguration` actually does not account for branches at all, as discussed [here](https://github.com/reposense/RepoSense/pull/547#issuecomment-470312239), which is why I only match the `RepoLocation`.

Reply:
Since branches are not accounted for `GroupConfiguration` as discussed [here](https://github.com/reposense/RepoSense/pull/547#issuecomment-470312239), I think it is sufficient to add 1 `RepoConfiguration` for each `RepoLocation`.